ThreadPool.h

```diff
@@ -62,25 +81,34 @@ inline void ThreadPool::setup(std::size_t threads, F&& initialize) {
         task();
       }
     });
+
+    pthread_attr_t pthread_attribute;
+    pthread_attr_init(&pthread_attribute);
+    pthread_attr_setstacksize(&pthread_attribute, stack_size);
+
+    pthread_create(&threads.back(), &pthread_attribute,
+                   &ThreadPool::entry_point, &entry_points.back());
+
+    pthread_attr_destroy(&pthread_attribute);
   }
 }
 
 // the constructor just launches some amount of workers and calls the
 // initializer in each
 template <class F>
-inline ThreadPool::ThreadPool(std::size_t threads, F&& initialize)
+inline ThreadPool::ThreadPool(std::size_t num_threads, F&& initialize,
+                              std::size_t stack_size)
     : stop(false) {
-  setup(threads, std::move(initialize));
+  setup(num_threads, std::forward<F>(initialize), stack_size);
 }
 
 // this constructor just launches the workers.
-inline ThreadPool::ThreadPool(std::size_t threads) : stop(false) {
-  setup(threads, []() {});
-}
+inline ThreadPool::ThreadPool(std::size_t num_threads)
+    : ThreadPool(num_threads, []() {}) {}
```

Comment:
this seems like a poor choice of constructor. never the less I've maintained the same functionality. I've chained the constructors to avoid copy/pasting